src/Donations/Controllers/DonationRequestController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Give\Donations\Controllers;
+
+use Give\Donations\Models\Donation;
+use Give\Donations\Repositories\DonationRepository;
+use Give\Donations\ValueObjects\DonationRoute;
+use Give\Framework\QueryBuilder\JoinQueryBuilder;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class DonationRequestController
+{
+    /**
+     * @unreleased
+     *
+     * @return WP_Error | WP_REST_Response
+     */
+    public function getDonation(WP_REST_Request $request)
+    {
+        $donation = Donation::find($request->get_param('id'));
+
+        if ( ! $donation) {
+            return new WP_Error('donation_not_found', __('Donation not found', 'give'), ['status' => 404]);
+        }
+
+        return new WP_REST_Response($donation->toArray());
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDonations(WP_REST_Request $request): WP_REST_Response
+    {
+        $page = $request->get_param('page');
+        $perPage = $request->get_param('per_page');
+
+        $query = give(DonationRepository::class)->prepareQuery();
+
+        if ($campaignId = $request->get_param('campaignId')) {
+            $query->distinct()->join(function (JoinQueryBuilder $builder) {
+                $builder->innerJoin('give_campaign_forms', 'campaign_forms')
+                    ->on('campaign_forms.form_id', "give_donationmeta_attach_meta_formId.meta_value");
+            })->where('campaign_forms.campaign_id', $campaignId);
+        }
+
+        $query
+            ->limit($perPage)
+            ->offset(($page - 1) * $perPage);
+
+        $donations = $query->getAll() ?? [];
+        $totalDonations = empty($donations) ? 0 : Donation::query()->count();
+        $totalPages = (int)ceil($totalDonations / $perPage);
+
+        $response = rest_ensure_response($donations);
+        $response->header('X-WP-Total', $totalDonations);
+        $response->header('X-WP-TotalPages', $totalPages);
```

Comment:
Cool to see headers and links used to add pagination! You should consider extracting this for re-use as a Friday project.